### PR TITLE
feat(server): immich checksum header

### PR DESCRIPTION
**Pull request by jrasm91**
check for `x-immich-checksum` and dedupe before handling the full file upload

### CODE REVIEW

File: server/src/utils/request.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export const fromChecksum = (checksum: string): Buffer => {
+  return Buffer.from(checksum, checksum.length === 28 ? 'base64' : 'hex');
```
Comment (danieldietzler) [inline server/src/utils/request.ts:2]: Why can it be guaranteed that base64 is always 28 character long? lol

**Review by danieldietzler — APPROVED** [review]

File: server/src/utils/request.ts (hunk 1 above)
Comment (jrasm91) [reply server/src/utils/request.ts:2]: https://stackoverflow.com/a/13378842

**Comment by benmccann** [comment]
Perhaps "hash" would be more accurate than "checksum"? I usually think of a checksum as being used for data integrity

**Comment by jrasm91** [comment]
> Perhaps "hash" would be more accurate than "checksum"? I usually think of a checksum as being used for data integrity

We could also use this for that in the future.

**Pull request merged**